From 20a68f5daad6ea7661246d91b00fdaf15b016144 Mon Sep 17 00:00:00 2001
From: Adrian Alonso <aalonso@freescale.com>
Date: Mon, 9 Apr 2012 12:18:08 -0500
Subject: [PATCH 1294/2463] ENGR00179230: imx-esai add record support for
 S24_LE format

* Add record support for S24_LE and
  S20_3LE bit format.

Signed-off-by: Adrian Alonso <aalonso@freescale.com>
---
 sound/soc/imx/imx-esai.c |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/sound/soc/imx/imx-esai.c b/sound/soc/imx/imx-esai.c
index f636171..22edbf9 100644
--- a/sound/soc/imx/imx-esai.c
+++ b/sound/soc/imx/imx-esai.c
@@ -375,6 +375,14 @@ static int imx_esai_hw_rx_params(struct snd_pcm_substream *substream,
 		rfcr |= ESAI_WORD_LEN_16;
 		rcr |= ESAI_RCR_RSHFD_MSB | ESAI_RCR_RSWS_STL32_WDL16;
 		break;
+	case SNDRV_PCM_FORMAT_S20_3LE:
+		rfcr |= ESAI_WORD_LEN_20;
+		rcr |= ESAI_RCR_RSHFD_MSB | ESAI_RCR_RSWS_STL32_WDL20;
+		break;
+	case SNDRV_PCM_FORMAT_S24_LE:
+		rfcr |= ESAI_WORD_LEN_24;
+		rcr |= ESAI_RCR_RSHFD_MSB | ESAI_RCR_RSWS_STL32_WDL24;
+		break;
 	}
 
 	channels = params_channels(params);
-- 
1.7.7.4

